zip-0226.rst

```diff
@@ -285,7 +285,9 @@ The value balances for each Asset Identifier in :math:`\mathsf{assetBurn}` repre
 
 The validator needs to verify the Balance and Binding Signature by adding the value balances for all Assets, as committed using their respective :math:`\mathsf{AssetBase}` as the value base point of the Pedersen Commitment. This is done as follows
 
-.. math:: \mathsf{bvk = (\sum cv_i^{net})}  - \mathsf{ ValueCommit_0^{Orchard}(v^{balanceOrchard})} - \sum_{\mathsf{assetBurn}} \mathsf{ValueCommit_0^{OrchardZSA}(AssetBase, v^{AssetBase}) } = \sum \mathsf{rcv_{i,j}^{net}}\mathcal{R}^{\mathsf{Orchard}}
+.. math:: \mathsf{bvk} = \mathsf{(\sum cv_i^{net})}  - \mathsf{ValueCommit_0^{Orchard}(v^{balanceOrchard})} - \sum_{\mathsf{assetBurn}} \mathsf{ValueCommit_0^{OrchardZSA}(AssetBase, v^{AssetBase})}
+    
+.. math:: \mathsf{bvk} = \sum \mathsf{rcv_{i,j}^{net}}\mathcal{R}^{\mathsf{Orchard}}
```

Comment:
AFAICT `rcv_{i,j}` is not standard notation (couldn't see any use of the double subscript in Zcash protocol specs) and the introduction of the `j` index is confusing. I suspect this notation is shortcut for `sum_{i=1}^{n} rvc_i + \sum_{j=1}^{m} rcv_j` or something like that, but I need to have a closer look at the specs and this part of the ZSA to fix this properly and mark this PR as "ready for review."

Reply:
Ok, so, I bounced on the Zcash specs to see if I was missing a definition or smthg.
AFAIK, there isn't such a thing as `rcv_{i,j}` so I would strongly suggest to either explicit what this means, or avoid to use this notation all together in order to remove any room for interpretation in the equations.

Looking at Orchard's specs (Section 4.14 Balance and Binding Signature (Orchard) of the Zcash specs):
- `bsk = \ssum^{n}_{i=1} rcv^{net}_i`
- `bvk = \csum^{n}_{i=1} cv^{net}_i \cminus ValueCommit_{0}(v^{balance})`

where:
- `\ssum` is the sum operation of the Pallas scalar field,
- `\csum` (resp. `\cminus`) is the sum (resp. minus) operation of the group (curve)
- `n` is the number of Action descriptions
- cv^net_{i} is the value commitment to the value of the input note minus the value v of the output note for ith Action transfer in the transaction, using rcv_i as random commitment trapdoor.

So, in this ZSA burn section, we build on Orchard's balance and binding signature check, but also allow the sender to include a set of tuples `(AssetBase, v^{AssetBase})` to burn `v^{AssetBase}` of asset `AssetBase`.

1) Out of curiosity: Why don't we use `assetId` instead, since `AssetBase` is derived from `AssetId`? Is that just to save on a lookup (or even more: save on the computation of ZSAValueBase) in the circuit? How is it implemented? I guess, the wallet handles the computation of `ZSAValueBase` to pass the asset's base point as input to the circuit? (If so, do we use caching if `ZSAValueBase` has already been computed for an assetId? In which case we just need to do a lookup I guess?)

Now looking at the current equation, we have:

```latex
\mathsf{bvk} = \mathsf{(\sum cv_i^{net})}  - \mathsf{ValueCommit_0^{Orchard}(v^{balanceOrchard})} - \sum_{\mathsf{assetBurn}} \mathsf{ValueCommit_0^{OrchardZSA}(AssetBase, v^{AssetBase})}

\mathsf{bvk} = \sum \mathsf{rcv_{i,j}^{net}}\mathcal{R}^{\mathsf{Orchard}}
```

So, I'll unroll the equations, adding explicit summation bounds, assuming we have `n` Action descriptions, and removing `\mathsf` for clarity.
Using the definition of `ValueCommit_0^{OrchardZSA}()` from this ZIP, we get:

```
bvk = (\sum^{n}_{i=1} cv_i^{net})  - ValueCommit_0^{Orchard}(v^{balanceOrchard}) - (\sum_{assetBurn} ValueCommit_0^{OrchardZSA}(AssetBase, v^{AssetBase}))

=> bvk =  (\sum^{n}_{i=1} cv_i^{net})  - ValueCommit_0^{Orchard}(v^{balanceOrchard}) - (\sum_{assetBurn} [v^{net}_{assetId}]AssetBase^{Orchard}_{assetId} + [0]R^{Orchard})

=> bvk =  (\sum^{n}_{i=1} [v^{net}_i]V^{Orchard} + [rcv_i^{net}]R^{Orchard})  - [v^{balanceOrchard}]V^{Orchard} + [0]R^{Orchard} - (\sum_{assetBurn} [v^{net}_{assetId}] AssetBase^{Orchard}_{assetId} + [0]R^{Orchard})
```

Now, I'm trying to understand how we lend on the equation:

```
bvk = \sum \mathsf{rcv_{i,j}^{net}}\mathcal{R}^{\mathsf{Orchard}}
```

So, let's assume that the cardinality of the `assetBurn` set is `m`. Getting back to our last equation, we can explicit the bounds of the sum in the last term to:

```
bvk =  (\sum^{n}_{i=1} [v^{net}_i]V^{Orchard} + [rcv_i^{net}]R^{Orchard})  - [v^{balanceOrchard}]V^{Orchard} + [0]R^{Orchard} - (\sum^{m}_{j=1} [v^{net}_{j}] AssetBase^{Orchard}_{j} + [0]R^{Orchard})
```

Note: we replaced `\sum_{assetBurn}` by `\sum^{m}_{j=1}`, we replaced `v^{net}_{assetId}` by `v^{net}_{j}` and we replaced `AssetBase^{Orchard}_{assetId}` by `AssetBase^{Orchard}_{j}` (following the assumption that there is no duplication in the set `assetBurn`, so there's a bijection between `assetBurn` and `{1..m}`).

From that point, and to the risk of missing something obvious, I don't see how we lend to `bvk = \sum \mathsf{rcv_{i,j}^{net}}\mathcal{R}^{\mathsf{Orchard}}` as currently stated in the ZIP. In fact, we pass 0 as random commitment trapdoor to all instances of `ValueCommit^{OrchardZSA}()` in this equation, so, unless I miss smthg, there is no such thing as `rcv_j`.

Reply:
AFAICT, `AssetBase` is derived via a Group Hash into Pallas (as per: https://qed-it.github.io/zips/protocol/protocol.pdf#concretegrouphashpallasandvesta), so I don't see any immediate relationship between `AssetBase`, `rcv` and `R^{Orchard}` that could lead to this `\sum rcv_{i,j}`.

Reply:
> Out of curiosity: Why don't we use assetId instead, since AssetBase is derived from AssetId? Is that just to save on a lookup (or even more: save on the computation of ZSAValueBase) in the circuit? How is it implemented? I guess, the wallet handles the computation of ZSAValueBase to pass the asset's base point as input to the circuit? (If so, do we use caching if ZSAValueBase has already been computed for an assetId? In which case we just need to do a lookup I guess?)

`AssetId` is long and can be up to 512B. We prefer to deal with the compact `AssetBase`. Especially important is the fact that the node need to keep track of all issued and finalized `AssetBase`/`AssetDigest`. So the idea is to strive for a small state as much as possible.

Reply:
We might need to introduce indexes in
`ValueCommit_0^{OrchardZSA}(AssetBase, v^{AssetBase}))` 
for `AssetBase`